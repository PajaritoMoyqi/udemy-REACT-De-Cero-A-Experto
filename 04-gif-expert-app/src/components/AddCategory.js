import React, { useState } from 'react';
import propTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputChange = e => {
    const val = e.target.value;

    setInputValue(val);
  }

  const handleSubmit = e => {
    e.preventDefault();

    if(inputValue.trim().length>2){
      setCategories(cats => [inputValue, ...cats]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input type='text' value={inputValue} onChange={handleInputChange} />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: propTypes.func.isRequired
}