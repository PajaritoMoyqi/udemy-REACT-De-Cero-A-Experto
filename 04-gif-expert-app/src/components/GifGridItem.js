import React from 'react'
import propTypes from 'prop-types';

export const GifGridItem = ({id, title, url}) => {

  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  ) // javascript class와 구분하고자 className을 쓴다.
}
GifGridItem.propTypes = {
  url: propTypes.string.isRequired,
  title: propTypes.string.isRequired
}