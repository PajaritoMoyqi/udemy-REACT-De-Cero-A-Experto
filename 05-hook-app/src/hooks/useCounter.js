import { useState } from "react"

export const useCounter = (initState = 10) => {

  const [state, setstate] = useState(initState);

  const increment = () => {
    setstate( state + 1 );
  }

  const decrement = () => {
    setstate( state - 1 );
  }

  return {
    counter: state, increment, decrement
  };

}
