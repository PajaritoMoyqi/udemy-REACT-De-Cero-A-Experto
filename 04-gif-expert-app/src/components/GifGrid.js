import React, {useState, useEffect} from 'react';
import propTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';



export const GifGrid = ({category}) => {
  // const [images, setImages] = useState([])

  // useEffect(()=>{
  //   getGifs(category)
  //     .then(setImages);
  // }, [category]); // []가 있어야 한 번만 실행된다. 처음 렌더링될 때만.
  // // category가 바뀌면 그때만 다시 실행한다.
  const {data:images, loading} = useFetchGifs(category);

  // getGifs(); // dom이 변할 때마다 계속 실행이 된다.

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && <p className="animate__animated animate__flash">'Cargando.....'</p>}

      <div className="card-grid">      
          {images.map(i => <GifGridItem key={i.id} {...i}/>)}
      </div>
    </>
  ) // list를 그대로 가감없이 보여주니까 그냥 map으로 안쪽 것들을 바꾸는 것뿐이다.
}

GifGrid.propTypes = {
  category: propTypes.string.isRequired
}