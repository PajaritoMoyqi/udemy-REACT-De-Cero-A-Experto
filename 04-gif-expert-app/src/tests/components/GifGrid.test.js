import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import React from "react";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('GifGrid test', () => {
  
  const category = 'One Punch'

  test('snapshot', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    }); // default values that useFetchGifs gives

    const wrapper = shallow(<GifGrid category={category} />)

    expect(wrapper).toMatchSnapshot();
  });

  test('show items when download img', () => {

    const gifs = [{
      id: 'abc',
      url: 'https://localhost/cucucu/cucu',
      title: 'vaina'
    },{
      id: 'abcd',
      url: 'https://localhost/cucucu/cucu',
      title: 'vainas'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    
    const wrapper = shallow(<GifGrid category={category} />)

    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length); // props number = length
  })
  
  test('should ', () => {
    
  })
  
})
