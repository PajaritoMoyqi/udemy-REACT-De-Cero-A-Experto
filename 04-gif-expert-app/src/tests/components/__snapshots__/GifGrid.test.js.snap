// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid test snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    'Cargando.....'
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
